.github/workflows/tests.yml

```diff
@@ -53,14 +55,51 @@ jobs:
 
       - name: Save Playwright Test HTML report
         if: always()
-        uses: actions/upload-artifact@v2
+        uses: actions/upload-artifact@master
         with:
           name: test-report-e2e
-          path: reports/index.html
+          path: allure-results
 
       - name: Save screenshots and trace in case of failure
         if: failure()
-        uses: actions/upload-artifact@v2
+        uses: actions/upload-artifact@master
         with:
           name: failure-screenshots-trace
-          path: test-results
+          path: allure-results
+  
+  generate_report:
+    name: Generate report
+    needs: [api_tests, e2e_tests]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download allure report
+        uses: actions/download-artifact@master
+        with:
+          name: allure-results
+
+      - name: Get allure history
+        uses: actions/checkout@v2
+        if: always()
+        continue-on-error: true
+        with:
+          ref: gh-pages
+          path: gh-pages
+
+      - name: Allure report action
+        uses: simple-elf/allure-report-action@master
+        if: always()
+        id: allure-report
+        with:
+          allure_results: allure-results
+          gh_pages: gh-pages
+          allure_report: allure-report
+          allure_history: allure-history
+          keep_reports: 20
```

Comment:
Let's keep the last 5, maybe? I don't know how much it will take from GHA artifact quota

Reply:
For sure.
No problems for me. 👯